db_checker.py

```diff
@@ -114,24 +116,20 @@ def allowed_unit(self, unit):
         """
         This method checks that the given unit conforms to standard
         """
-        if unit in allowed_units:
-            return True
+        # expand macro $(A) to a valid unit, expand $(A=B) to B
+        u = re.sub(r'\$[({].*?=(.*)?[})]', r'\1', unit)
+        u = re.sub(r'\$[({].*?[})]', 'm', u)
 
-        # otherwise check for macro
-        if "$" in unit:
-            return True
+        # split unit amalgamations
+        units = re.split(r'[/ ()]', u)
 
-        # otherwise split unit amalgamations
-        units = re.split(r'[/ ()]', unit)
         for u in units:
+            # remove any powers of units e.g. mm^2 -> mm
+            u = re.sub(r'\^[-]?\d', '', u)
```

Comment:
This does not check whether the power symbol is actually in the right place and passes e.g. ^2MHz, M^2H^2z etc. Also allows endlessly stacking power units e.g. MHz^2^-3^2^3, which although technically a valid unit, I'm not sure whether it's a sensible thing to allow?